spec/features/user_can_edit_profile_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+describe "user can edit profile" do
+  before :each do
+    User.create(username: "dave", password: "123", role: 0)
+    visit root_url
+    click_link "Login"
+    fill_in("Username", with: "dave")
+    fill_in("Password", with: "123")
+    click_button("Login")
+  end
+
+  it "can edit the profile" do
+    visit '/profile'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.